src/braket/pennylane_plugin/braket_device.py

```diff
@@ -355,6 +366,33 @@ def apply(
 
         return circuit
 
+    def _parametrized_evolution_to_pulse_sequence(
+        self, op: ParametrizedEvolution, dev_wires
+    ) -> PulseSequence:
+        pulse = op.H.pulses[0]  # Assume only one `HardwarePulse`
+        amplitude = partial(pulse.amplitude, op.parameters[0])
+
+        frames = [self._device.frames[f"q{w}_drive"] for w in dev_wires]
+        sampling_periods = [f.port.dt * 1e9 for f in frames]  # seconds to nanoseconds
+
+        # TODO: Figure out unit of amplitude for `ArbitraryWaveform` and do conversion
+        amplitudes = [
+            [amplitude(t) * 1e9 for t in np.arange(op.t[0], op.t[1], dt)] for dt in sampling_periods
```

Comment:
Also needs to handle constants differently but I guess you are planning to add the `ConstantWaveform` anyway